modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala

```diff
@@ -0,0 +1,40 @@
+package org.tessellation.sdk.domain.healthcheck.services
+
+import cats.effect.Ref
+import cats.syntax.bifunctor._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+import cats.{Applicative, Monad}
+
+import org.tessellation.sdk.domain.healthcheck.types.ConsensusRounds
+
+abstract class HealthCheckConsensus[F[_]: Monad]() extends HealthCheck[F] {
+  def allRounds: Ref[F, ConsensusRounds[F]]
+  def roundsInProgress = allRounds.get.map(_.inProgress)
+
+  final override def trigger(): F[Unit] =
+    triggerRound()
+
+  def triggerRound(): F[Unit] =
+    roundsInProgress.flatMap { inProgress =>
+      if (inProgress.nonEmpty) manageRounds(inProgress) else Applicative[F].unit
+    }
+
+  def manageRounds(rounds: ConsensusRounds.InProgress[F]): F[Unit] =
+    partition(rounds).flatMap {
+      case (finished, inProgress) => Applicative[F].unit
+    }
```

Comment:
`partition(rounds).void` unless you will do something with `finished` and `inProgress` here

Reply:
I will :) I said it's a stub/dummy code for now